browser/search_engines/tor_window_search_engine_provider_service.cc

```diff
@@ -15,8 +16,8 @@
 TorWindowSearchEngineProviderService::
 TorWindowSearchEngineProviderService(Profile* otr_profile)
     : SearchEngineProviderService(otr_profile) {
-  DCHECK(otr_profile->IsTorProfile());
-  DCHECK(IsGuestProfile(otr_profile));
+  DCHECK(tor::IsTorProfile(otr_profile));
+  DCHECK(otr_profile->IsGuestSession());
```

Comment:
I don't think this works here. In IsGuestSession it's `is_guest_profile_ || tor::IsTorProfile(this)` and is_guest_profile_ is still false at this time, so it falls through IsTorProfile and doesn't DCHECK.

Reply:
It’s actually a leftover when I moved IsGuestSession to IsOTRGuestProfile. (I removed IsGuestSession first and then I saw your comment in original PR)
I will force push an updated commit

Reply:
addressed in 8200e2b2d8b0fbdbdc3aeae4e17154703da746d5